crowbar_framework/app/controllers/installers_controller.rb

```diff
@@ -42,27 +46,74 @@ def status
   # Provides the restful api call for
   # /installer/start 	POST 	triggers installation
   def start
-    if Crowbar::Installer.successful?
-      respond_to do |format|
-        format.json { render json: Crowbar::Installer.status }
-        format.html { redirect_to installer_url }
-      end
-    else
-      # the shell Process will be spawned in the background and therefore has
-      # not a direct return value which we can use here
-      if Crowbar::Installer.installing?
-        flash[:notice] = I18n.t(".installers.start.installation_ongoing")
-      else
-        ret = Crowbar::Installer.install
-        case ret[:status]
-        when 501
+    if params[:force]
+      ret = Crowbar::Installer.install!
+      case ret[:status]
+      when 501
+        respond_to do |format|
+          format.json do
+            head :not_implemented
+          end
+        end
+        format.html do
           flash[:alert] = ret[:msg]
+          redirect_to installer_url
         end
       end
 
       respond_to do |format|
-        format.json { head :ok }
-        format.html { redirect_to installer_url }
+        format.json do
+          head :ok
+        end
+        format.html do
+          redirect_to installer_url
+        end
+      end
+    else
+      if Crowbar::Installer.successful?
+        respond_to do |format|
+          format.json do
+            head :gone
+          end
+          format.html do
+            redirect_to installer_url
+          end
+        end
+      else
+        if Crowbar::Installer.installing?
+          respond_to do |format|
+            format.json do
+              head :im_used
+            end
+            format.html do
+              flash[:notice] = I18n.t(".installers.start.installation_ongoing")
+              redirect_to installer_url
+            end
+          end
+        else
+          ret = Crowbar::Installer.install
+          case ret[:status]
```

Comment:
Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)